### ISSUE TITLE
Probability

### ISSUE THREAD

**Issue by prince14322** [xuxingya/tf2crf#3]
Instead of one hot, how to return the probabilities of classes?

**Comment by xuxingya** [comment]
If you want the probabilities, you don't need a CRF Layer, use a Dense Layer with 'softmax'.
```
from tensorflow.keras.layers import Input, Embedding, Bidirectional, GRU, Dense
from tensorflow.keras.models import Model

inputs = Input(shape=(None,), dtype='int32')
output = Embedding(len(vocab), dim, trainable=True, mask_zero=True)(inputs)
output = Bidirectional(GRU(64, return_sequences=True))(output)
output = Dense(len(class_num), activation='softmax')(output)
model = Model(inputs, output)
model.compile(loss='sparse_categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
```
You use 'sparse_categorical_crossentropy' loss for sparse Y and 'categorical_crossentropy' loss for one-hot encoding Y.

**Comment by prince14322** [comment]
using CRF layer, how to compute probabilities?

**Comment by xuxingya** [comment]
I don't know what you mean by computing probabilities. The CRF is used to
find the most possible sequence according to the probabilities of tokens
and transfer probabilities. The output of CRF
 is always indices.
